bootstrap/api/v1beta1/rke2config_webhook.go

```diff
@@ -158,10 +158,34 @@ func (s *RKE2ConfigSpec) validate(pathPrefix *field.Path) field.ErrorList {
 
 	allErrs = append(allErrs, s.validateIgnition(pathPrefix)...)
 	allErrs = append(allErrs, s.validateRegistries(pathPrefix)...)
+	allErrs = append(allErrs, s.validateFileSource(pathPrefix)...)
 
 	return allErrs
 }
 
+// validateFileSource ensures that either ConfigMap or Secret FileSource is provided; it raises error if both or none are provided
+func (s *RKE2ConfigSpec) validateFileSource(pathPrefix *field.Path) field.ErrorList {
```

Comment:
out of the scope for this PR but we might want use CEL instead of admission webhooks